File: src/enums.py
from enum import Enum


class RequestDatabaseType(str, Enum):
    main = "main"
    suite = "suite"

    def __str__(self) -> str:
        return self.value


class SekaiBindingServerRegion(str, Enum):
    jp = "jp"
    en = "en"
    tw = "tw"
    kr = "kr"
    cn = "cn"
    default = "default"

    def __str__(self):
        return self.value


class SekaiSuiteDataType(str, Enum):
    suite = "suite"
    mysekai = "mysekai"

    def __str__(self) -> str:
        return self.value


class InstantMessengerPlatform(str, Enum):
    qq = "qq"
    qq_bot = "qq_bot"
    discord = "discord"
    telegram = "telegram"

    def __str__(self):
        return self.value


class PjskAliasType(str, Enum):
    music = "music"
    character = "character"

    def __str__(self):
        return self.value

File: utils/model/enums.go
package model

// RequestDatabaseType represents the type of database to request
type RequestDatabaseType string

const (
	RequestDatabaseTypeMain  RequestDatabaseType = "main"
	RequestDatabaseTypeSuite RequestDatabaseType = "suite"
)

func (r RequestDatabaseType) String() string {
	return string(r)
}

// SekaiBindingServerRegion represents the server region for Sekai binding
type SekaiBindingServerRegion string

const (
	SekaiBindingServerRegionJP      SekaiBindingServerRegion = "jp"
	SekaiBindingServerRegionEN      SekaiBindingServerRegion = "en"
	SekaiBindingServerRegionTW      SekaiBindingServerRegion = "tw"
	SekaiBindingServerRegionKR      SekaiBindingServerRegion = "kr"
	SekaiBindingServerRegionCN      SekaiBindingServerRegion = "cn"
	SekaiBindingServerRegionDefault SekaiBindingServerRegion = "default"
)

func (s SekaiBindingServerRegion) String() string {
	return string(s)
}

// SekaiSuiteDataType represents the type of Sekai suite data
type SekaiSuiteDataType string

const (
	SekaiSuiteDataTypeSuite   SekaiSuiteDataType = "suite"
	SekaiSuiteDataTypeMySekai SekaiSuiteDataType = "mysekai"
)

func (s SekaiSuiteDataType) String() string {
	return string(s)
}

// InstantMessengerPlatform represents the instant messenger platform
type InstantMessengerPlatform string

const (
	InstantMessengerPlatformQQ       InstantMessengerPlatform = "qq"
	InstantMessengerPlatformQQBot    InstantMessengerPlatform = "qq_bot"
	InstantMessengerPlatformDiscord  InstantMessengerPlatform = "discord"
	InstantMessengerPlatformTelegram InstantMessengerPlatform = "telegram"
)

func (i InstantMessengerPlatform) String() string {
	return string(i)
}

// PjskAliasType represents the type of PJSK alias
type PjskAliasType string

const (
	PjskAliasTypeMusic     PjskAliasType = "music"
	PjskAliasTypeCharacter PjskAliasType = "character"
)

func (p PjskAliasType) String() string {
	return string(p)
}
